htdocs/modules/system/admin/blocksadmin/main.php

```diff
@@ -279,7 +279,7 @@
             $name = $block->getVar('name');
             // Save block options
             $options = $_POST['options'];
-            if (isset($options)) {
+            if (isset($options) && is_array($options)) {
```

Comment:
Should check to see if `$_POST['options']` is set, not `$options`.  `$options` was assigned in the previous statement.
```suggestion
            $options = isset($_POST['options']) ? $_POST['options'] : [];
            if (is_array($options) && !empty($options)) {
```

Reply:
Thanks! How about: 
$options = Xmf\Request::getArray('options', [], 'POST');
if (is_array($options) && !empty($options)) {